### PR TITLE
Update order status from magento

### PR DESCRIPTION
**Pull request by prakashpp**

### CODE REVIEW

File: channel.py [hunk 1]
```diff
@@ -668,6 +668,19 @@ def get_default_tryton_action(self, name):
                 'shipment_method': 'manual'
             }
 
+    def update_order_status(self):
+        Sale = Pool().get('sale.sale')
+
+        if self.source != 'magento':
+            super(Channel, self).update_order_status()
```
Comment (tarunbhardwaj) [inline channel.py]: `return ...`

File: sale.py [hunk 2]
```diff
@@ -514,6 +514,35 @@ def copy(cls, sales, default=None):
...
+        with magento.Order(
+            self.channel.magento_url, self.channel.magento_api_user,
+            self.channel.magento_api_key
+        ) as order_api:
+            order_data = order_api.info(self.reference)
+
+        if order_data['status'] == 'complete':
+            # Order is completed on magento, process shipments and
+            # invoices.
+            for shipment in self.shipments:
```
Comment (tarunbhardwaj) [inline sale.py]: Shipments will only be available after processing state of sale. so either use this method for processing state sales only or if sale is not in processing state, then process sale first ?

Comment (prakashpp) [inline sale.py]: default order mapping import order in processing state, if some reason order is not in process state then people could have fixed the issue and make it processed?

File: channel.py [hunk 3]
```diff
@@ -668,6 +668,19 @@ def get_default_tryton_action(self, name):
...
 
+    def update_order_status(self):
+        Sale = Pool().get('sale.sale')
+
+        if self.source != 'magento':
+            return super(Channel, self).update_order_status()
+
+        sales, = Sale.search([
+            ('channel', '=', self.id),
+            ('state', 'not in', ('done', 'canceled')),
```
Comment (sharoonthomas) [inline channel.py]: i would prefer a positive in search

`('state', 'in', ['confirmed', 'processing'])`

File: sale.py (hunk 2 above)
Comment (sharoonthomas) [inline sale.py]: @prakashpp that is only a default, could be different in the user's case

**Pull request merged**